aloha-io-avro/src/main/scala/com/eharmony/aloha/factory/avro/factory_config.scala

```diff
@@ -0,0 +1,165 @@
+  *
+  * This is especially useful for creating factories in generic services because the
+  * `modelCodomainRefInfoStr` is a string rather than `RefInfo` so it can come from a
+  * property file.
+  *
+  * @param schema an Avro Schema that represents the data passed to models created by this factory.
+  * @param modelCodomainRefInfoStr A string representation of a `com.eharmony.aloha.reflect.RefInfo`.
+  * @param imports imports to be injected into feature functions synthesized by the factory.
+  * @param classCacheDir a cache directory on the local machine used to cache class files of
+  *                      the created feature functions used in the models produced by the
+  *                      factory.
+  * @param dereferenceAsOptional whether to treat the dereferencing of repeated variables as
+  *                              an optional type.  This avoids index out of bounds exceptions
+  *                              and is safer but slightly slower.
+  */
+case class SchemaConfig(
+    schema: Schema,
+    modelCodomainRefInfoStr: String,
+    imports: Seq[String] = Nil,
+    classCacheDir: Option[File] = None,
+    dereferenceAsOptional: Boolean = true
+) extends FactoryConfig {
+
+  /**
+    * @return A Try of a ModelFactory that creates models taking `GenericRecord` instances as
+    *         input and returns `com.eharmony.aloha.audit.impl.avro.Score` as output.
+    */
+  def apply(): Try[ModelFactory[GenericRecord, Score]] = {
+    val f = for {
+      ri <- refInfo(modelCodomainRefInfoStr)
+      a <- auditor(ri)
+      s = semantics(schema, imports, classCacheDir, dereferenceAsOptional)
+      mf = ModelFactory.defaultFactory(s, a)(ri)
+    } yield mf
+
+    FactoryConfig.wrapException(f)
+  }
+}
+
+/**
+  * Provides an easy interface for creating ModelFactory instances that both take and
+  * return Avro objects.
+  *
+  * This is especially useful for creating factories in generic services because the
+  * `modelCodomainRefInfoStr` is a string rather than `RefInfo` so it can come from a
+  * property file.
+  *
+  * @param vfs A [[com.eharmony.aloha.io.vfs.Vfs]] wrapper around a URL pointing to a JSON Avro
+  *            Schema that represents the data passed to models created by this factory.
+  * @param modelCodomainRefInfoStr A string representation of a `com.eharmony.aloha.reflect.RefInfo`.
+  * @param imports imports to be injected into feature functions synthesized by the factory.
+  * @param classCacheDir a cache directory on the local machine used to cache class files of
+  *                      the created feature functions used in the models produced by the
+  *                      factory.
+  * @param dereferenceAsOptional whether to treat the dereferencing of repeated variables as
+  *                              an optional type.  This avoids index out of bounds exceptions
+  *                              and is safer but slightly slower.
+  */
+case class UrlConfig(
+    vfs: Vfs,
```

Comment:
@jmorra This is a big change but I think it's for the best.  It makes things a bit more verbose when defining but I think it's worth it.  I could make the `Vfs1` and `Vfs2` factory methods in `c.e.a.io.vfs` implicit to save a little. 

```scala
object Vfs1 {
  implicit def apply(fo: vfs1.FileObject): Vfs1 = ???
}
```

```scala
object Vfs2 {
  implicit def apply(fo: vfs1.FileObject): Vfs2 = ???
}
```

What do you think?